generic_config_updater/field_operation_validators.py

```diff
@@ -126,3 +126,71 @@ def _get_fields_in_patch():
                 return False
 
     return True
+
+
+def read_statedb_entry(table, field):
+    state_db = SonicV2Connector(host="127.0.0.1")
+    state_db.connect(state_db.STATE_DB)
+    return state_db.get(state_db.STATE_DB, table, field)
+
+
+def port_config_update_validator(patch_element):
+
+    def _validate_key(key, port, value):
+        if key == "fec":
+            supported_fecs_str = read_statedb_entry('{}|{}'.format("PORT_TABLE", port), "supported_fecs")
+            if supported_fecs_str:
+                if supported_fecs_str != 'N/A':
+                    supported_fecs_list = [element.strip() for element in supported_fecs_str.split(',')]
+                else:
+                    supported_fecs_list = []
+            else:
+                supported_fecs_list = [ 'rs', 'fc', 'none']
+            if value.strip() not in supported_fecs_list:
+                return False
+            return True
+        if key == "speed":
+            supported_speeds_str = read_statedb_entry('{}|{}'.format("PORT_TABLE", port), "supported_speeds") or ''
+            supported_speeds = [int(s) for s in supported_speeds_str.split(',') if s]
+            if supported_speeds and int(value) not in supported_speeds:
+                return False
+            return True
```

Comment:
>return [](http://example.com/codeflow?start=12&length=6)

Could you add a final `return false` at the end of function? The default behavior is `return None`, which is working but not reader friendly.

Reply:
yes, I changed to `return False`